### ISSUE TITLE
Model generated with blueprint doesn't work (with Octane preview)

### ISSUE THREAD

**Issue by pgengler** [emberjs/data#5937]
### Reproduction
https://github.com/pgengler/octane-preview-app/tree/attr-decorators-for-ember-data-models

(Start `ember s` and visit http://localhost:4200/tests to see this in action.)

### Description
Declaring an attribute on a model using the `@attr` decorator doesn't actually set up that attribute on the model if used as a bare decorator; using `@attr()` (or `@attr('string')`, e.g.) seems to be ok.

On a model generated using `ember g model item value`:
```js
import DS from 'ember-data';
const { Model, attr } = DS;

export default class ItemModel extends Model {
  @attr value;
}
```

The error I get from doing `this.store.createRecord('item', { value: 'abc' })` is:
```
TypeError: Cannot assign to read only property 'value' of object '[object Object]'
    at defineProperty (http://localhost:4200/assets/vendor.js:14812:22)
    at initialize (http://localhost:4200/assets/vendor.js:29517:41)
    at Function.create (http://localhost:4200/assets/vendor.js:30060:9)
    at FactoryManager.create (http://localhost:4200/assets/vendor.js:1215:33)
    at InternalModel.getRecord (http://localhost:4200/assets/vendor.js:69232:63)
    at http://localhost:4200/assets/vendor.js:76149:32
    at Backburner._run (http://localhost:4200/assets/vendor.js:53686:25)
    at Backburner._join (http://localhost:4200/assets/vendor.js:53662:21)
    at Backburner.join (http://localhost:4200/assets/vendor.js:53379:19)
    at http://localhost:4200/assets/vendor.js:76130:33
```

### Versions

```cli
$ npm ls ember-source && npm ls ember-cli && npm ls ember-data
octane-app@0.0.0 /Users/pgengler/octane-app
└── ember-source@3.10.0-canary

octane-app@0.0.0 /Users/pgengler/octane-app
└── ember-cli@3.9.0-beta.2  (github:ember-cli/ember-cli#ed51787b3e18efee706f163d57836a30cd5286b7)

octane-app@0.0.0 /Users/pgengler/octane-app
└── ember-data@3.9.0-canary  (github:emberjs/data#435ad9d8b92998fcbe2cee1746612148648280e1)
```

**Comment by runspired** [comment]
cc @pzuraq

**Comment by pzuraq** [comment]
@pgengler You have to use `@attr()` with parenthesis for new. Ember Data itself will have to update to provide a decorator default that doesn't require the parenthesis, essentially it'll have to detect if it's being called as a decorator or not, and if so, call the function internally with no parameters.

**Comment by runspired** [comment]
@pzuraq any guidance on how to detect that?

**Comment by runspired** [comment]
Fixed by #6054
